features/fixtures/ios-swift-cocoapods/iOSTestApp/AppDelegate.swift

```diff
@@ -45,11 +36,30 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         }
         assert(mockAPIPath.count > 0, "The mock API path must be set prior to triggering events")
 
+        let config = prepareConfig(apiKey: bugsnagAPIKey, mockAPIPath: mockAPIPath)
+        let scenario = scenarioForEventType(eventType: eventType, config: config)
+        triggerEvent(scenario: scenario, delay: delay)
+    }
+
+    internal func prepareConfig(apiKey: String, mockAPIPath: String) -> BugsnagConfiguration {
         let config = BugsnagConfiguration()
-        config.apiKey = bugsnagAPIKey
-        config.notifyURL = URL(string: mockAPIPath)
-        Bugsnag.start(with: config)
+        let url = URL(string: mockAPIPath)
+        config.apiKey = apiKey
+        config.notifyURL = url
+        config.sessionURL = url
+        return config
+    }
+
+    internal func scenarioForEventType(eventType: String, config: BugsnagConfiguration) -> Scenario {
+        let eventTypeForClass = "iOSTestApp." + eventType // prefix with app name
+        let type = NSClassFromString(eventTypeForClass) as! Scenario.Type
+        return type.init(config: config)
+    }
 
-        triggerEvent(name: eventType, delay: delay)
+    func triggerEvent(scenario: Scenario, delay: TimeInterval) {
+        let when = DispatchTime.now() + delay
+        DispatchQueue.main.asyncAfter(deadline: when) {
```

Comment:
Given the scenario structure, I'd move the delay into the `run` method because otherwise Bugsnag is initialized after the delay, which might cause some interesting edge cases.

Reply:
I've added an `initBugsnag` method which is called prior to `run`.